modules/analyze_basic_content_info/src/Plugin/Analyze/ContentInfo.php

```diff
@@ -0,0 +1,190 @@
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Analyze plugin to display Basic data.
+ *
+ * @Analyze(
+ *   id = "content_info",
+ *   label = @Translation("Basic Content Entity Reports"),
+ *   description = @Translation("Provides basic statistics about content for Analyzer.")
+ * )
+ */
+final class ContentInfo extends AnalyzePluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Creates the plugin.
+   *
+   * @param array $configuration
+   *   Configuration.
+   * @param $plugin_id
+   *   Plugin ID.
+   * @param $plugin_definition
+   *   Plugin Definition.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entityFieldManager
+   *   Statistics service.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    protected EntityFieldManagerInterface $entityFieldManager,
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function renderSummary(EntityInterface $entity): array {
+    return [
+      '#type' => 'table',
+      '#header' => [['data' => 'Basic Info', 'colspan' => 2, 'class' => ['header']]],
+      '#rows' => [
+        ['data' => ['Word count', $this->getWordCount($entity)]],
+        ['data' => ['Image count', $this->getImageCount($entity)]],
+      ],
+      '#attributes' => [
+        'class' => ['basic-data-table'],
+        'style' => ['table-layout: fixed;'],
+      ],
+    ];
+  }
+
+  /**
+   * Helper to calculate a rough word count for the entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to calculate a word count for.
+   *
+   * @return int
+   *   The counted words. Defaults to 0.
+   */
+  private function getWordCount(EntityInterface $entity): int {
+    $return = 0;
+    $countables = [];
+
+    if ($entity instanceof FieldableEntityInterface) {
+      foreach ($this->getFieldDefinitionsByType($entity, [
+        'string',
+        'string_long',
+        'text_long',
+        'text_with_summary',
+        'text',
+      ]) as $definition) {
+        if ($name = $definition->getName()) {
+          if ($name !== 'timezone') {
+            $value = $entity->get($name)->getValue();
+
+            if (!empty($value[0]['value'])) {
+              $countables[] = $value[0]['value'];
+            }
+          }
+        }
+      }
+    }
+    else {
+
+      // Handle potential differences in how titles are collected.
+      if (method_exists($entity, 'getTitle')) {
```

Comment:
Never seen this one before! What uses this? `EntityInterface::label` should be available for all entities?

Reply:
resolved